{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8fbf475e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6ed98d35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2117, 15)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combo_metadata = pd.read_csv('../data/Metabolome/26112024/Metadata_all.tsv', sep='\\t', index_col=0)\n",
    "combo_metadata.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "54eb8d60",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_3xtg_fecal = pd.read_csv('../data/JointRPCA/3XTG_sac_fecal_metab-v-micro_correlation_table/Correlation.tsv',\n",
    "                              sep='\\t', index_col=0)\n",
    "corr_3xtg_brain = pd.read_csv('../data/JointRPCA/3XTG_sac_brain-v-micro_correlation_table/Correlation.tsv',\n",
    "                              sep='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "267d96d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_distance_matrix(distance_matrix, row_filter, col_filter):\n",
    "    \"\"\"\n",
    "    Filters a distance matrix based on row and column prefixes.\n",
    "\n",
    "    Args:\n",
    "        distance_matrix: A pandas DataFrame representing the distance matrix.\n",
    "        row_filter: The prefix to filter rows by.\n",
    "        col_filter: The prefix to filter columns by.\n",
    "\n",
    "    Returns:\n",
    "        A filtered pandas DataFrame.\n",
    "    \"\"\"\n",
    "\n",
    "    row_filter_mask = distance_matrix.index.str.startswith(row_filter)\n",
    "    col_filter_mask = distance_matrix.columns.str.startswith(col_filter)\n",
    "\n",
    "    filtered_matrix = distance_matrix.loc[row_filter_mask, col_filter_mask]\n",
    "\n",
    "    return filtered_matrix\n",
    "\n",
    "def transform_df(df):\n",
    "    \"\"\"\n",
    "    Transforms a DataFrame to have the index as the first column, \n",
    "      column names as the second column, and values in a new column.\n",
    "\n",
    "    Args:\n",
    "        df: The input DataFrame.\n",
    "\n",
    "    Returns:\n",
    "        A new DataFrame with the transformed structure.\n",
    "    \"\"\"\n",
    "        \n",
    "    idx = []\n",
    "    cols = []\n",
    "    val = []\n",
    "    for index, row in df.iterrows():\n",
    "        for col, value in row.items():\n",
    "            idx.append(index)\n",
    "            cols.append(col)\n",
    "            val.append(value)\n",
    "\n",
    "      # Create the new DataFrame\n",
    "    new_df = pd.DataFrame({'OGU':idx, 'metab_row_id':cols, 'jointRPCA_co-occur_value':val})\n",
    "    \n",
    "    return new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f799ee64",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1952, 5626)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_3xtg_fecal_filt = filter_distance_matrix(corr_3xtg_fecal, \n",
    "                                              row_filter=\"G\", \n",
    "                                              col_filter=\"fecal_\") \n",
    "corr_3xtg_fecal_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "97738114",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005825-64</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>64</td>\n",
       "      <td>0.786478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-319</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>319</td>\n",
       "      <td>0.298362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-2065</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>2065</td>\n",
       "      <td>0.984805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-459</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>459</td>\n",
       "      <td>0.678559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-83</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>83</td>\n",
       "      <td>-0.241517</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        OGU  metab_row_id  jointRPCA_co-occur_value\n",
       "node                                                               \n",
       "G000005825-64    G000005825            64                  0.786478\n",
       "G000005825-319   G000005825           319                  0.298362\n",
       "G000005825-2065  G000005825          2065                  0.984805\n",
       "G000005825-459   G000005825           459                  0.678559\n",
       "G000005825-83    G000005825            83                 -0.241517"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cyscp_3xtg_fecal = transform_df(corr_3xtg_fecal_filt)\n",
    "cyscp_3xtg_fecal['metab_row_id'] = cyscp_3xtg_fecal['metab_row_id'].str.replace('fecal_spf_3xtg_', '')\n",
    "cyscp_3xtg_fecal['metab_row_id'] = cyscp_3xtg_fecal['metab_row_id'].astype(int)\n",
    "cyscp_3xtg_fecal['node']= cyscp_3xtg_fecal['OGU'].astype(str)+\"-\"+cyscp_3xtg_fecal['metab_row_id'].astype(str)\n",
    "cyscp_3xtg_fecal.set_index('node', inplace=True)\n",
    "cyscp_3xtg_fecal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "353b26c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cyscp_3xtg_fecal.to_csv('../data/JointRPCA/3xtg_sac_fecal_cytoscape_format.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "17ca3360",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    6.574987e+06\n",
       "mean     2.717109e-02\n",
       "std      8.046525e-01\n",
       "min     -1.000000e+00\n",
       "25%     -7.959281e-01\n",
       "50%      5.189001e-01\n",
       "75%      8.273432e-01\n",
       "max      9.999999e-01\n",
       "Name: jointRPCA_co-occur_value, dtype: float64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_fecal = cyscp_3xtg_fecal[np.abs(cyscp_3xtg_fecal['jointRPCA_co-occur_value']) > 0.5].copy()\n",
    "short_fecal['jointRPCA_co-occur_value'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "70a01d9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005825-64</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>64</td>\n",
       "      <td>0.786478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-2065</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>2065</td>\n",
       "      <td>0.984805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-459</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>459</td>\n",
       "      <td>0.678559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-961</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>961</td>\n",
       "      <td>0.788988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-1693</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>1693</td>\n",
       "      <td>-0.722251</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        OGU  metab_row_id  jointRPCA_co-occur_value\n",
       "node                                                               \n",
       "G000005825-64    G000005825            64                  0.786478\n",
       "G000005825-2065  G000005825          2065                  0.984805\n",
       "G000005825-459   G000005825           459                  0.678559\n",
       "G000005825-961   G000005825           961                  0.788988\n",
       "G000005825-1693  G000005825          1693                 -0.722251"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_fecal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cef5a46f",
   "metadata": {},
   "outputs": [],
   "source": [
    "short_fecal.to_csv('../data/JointRPCA/3xtg_sac_fecal_cytoscape_format_cutoff50.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "55eacb93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1902, 890)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_3xtg_brain_filt = filter_distance_matrix(corr_3xtg_brain, \n",
    "                                              row_filter=\"G\", \n",
    "                                              col_filter=\"brain_\") \n",
    "corr_3xtg_brain_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "febab659",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005825-532</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>532</td>\n",
       "      <td>0.035114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-703</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>703</td>\n",
       "      <td>0.360000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-816</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>816</td>\n",
       "      <td>-0.007822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-881</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>881</td>\n",
       "      <td>0.197130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-1017</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>1017</td>\n",
       "      <td>0.345076</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        OGU  metab_row_id  jointRPCA_co-occur_value\n",
       "node                                                               \n",
       "G000005825-532   G000005825           532                  0.035114\n",
       "G000005825-703   G000005825           703                  0.360000\n",
       "G000005825-816   G000005825           816                 -0.007822\n",
       "G000005825-881   G000005825           881                  0.197130\n",
       "G000005825-1017  G000005825          1017                  0.345076"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cyscp_3xtg_brain = transform_df(corr_3xtg_brain_filt)\n",
    "cyscp_3xtg_brain['metab_row_id'] = cyscp_3xtg_brain['metab_row_id'].str.replace('brain_', '')\n",
    "cyscp_3xtg_brain['metab_row_id'] = cyscp_3xtg_brain['metab_row_id'].astype(int)\n",
    "cyscp_3xtg_brain['node']= cyscp_3xtg_brain['OGU'].astype(str)+\"-\"+cyscp_3xtg_brain['metab_row_id'].astype(str)\n",
    "cyscp_3xtg_brain.set_index('node', inplace=True)\n",
    "cyscp_3xtg_brain.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0b04da32",
   "metadata": {},
   "outputs": [],
   "source": [
    "cyscp_3xtg_brain.to_csv('../data/JointRPCA/3xtg_brain_cytoscape_format.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d0d6770b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    1.692780e+06\n",
       "mean     1.797493e-02\n",
       "std      5.856040e-01\n",
       "min     -9.999997e-01\n",
       "25%     -4.839313e-01\n",
       "50%      9.210277e-03\n",
       "75%      5.329324e-01\n",
       "max      9.999999e-01\n",
       "Name: jointRPCA_co-occur_value, dtype: float64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cyscp_3xtg_brain['jointRPCA_co-occur_value'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e0cc5e66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1692780, 3)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cyscp_3xtg_brain.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "44249112",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(859400, 3)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cyscp_3xtg_brain[np.abs(cyscp_3xtg_brain['jointRPCA_co-occur_value']) > 0.5].shape\n",
    "#reduced by half"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "59df9654",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    859400.000000\n",
       "mean          0.041586\n",
       "std           0.772046\n",
       "min          -1.000000\n",
       "25%          -0.740169\n",
       "50%           0.524789\n",
       "75%           0.785735\n",
       "max           1.000000\n",
       "Name: jointRPCA_co-occur_value, dtype: float64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_brain = cyscp_3xtg_brain[np.abs(cyscp_3xtg_brain['jointRPCA_co-occur_value']) > 0.5].copy()\n",
    "short_brain['jointRPCA_co-occur_value'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f5f7c17a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005825-871</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>871</td>\n",
       "      <td>0.965340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-440</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>440</td>\n",
       "      <td>0.855243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-928</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>928</td>\n",
       "      <td>0.600555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-659</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>659</td>\n",
       "      <td>-0.759782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-851</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>851</td>\n",
       "      <td>0.742475</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       OGU  metab_row_id  jointRPCA_co-occur_value\n",
       "node                                                              \n",
       "G000005825-871  G000005825           871                  0.965340\n",
       "G000005825-440  G000005825           440                  0.855243\n",
       "G000005825-928  G000005825           928                  0.600555\n",
       "G000005825-659  G000005825           659                 -0.759782\n",
       "G000005825-851  G000005825           851                  0.742475"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_brain.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "baa4b061",
   "metadata": {},
   "outputs": [],
   "source": [
    "short_brain.to_csv('../data/JointRPCA/3xtg_brain_cytoscape_format_cutoff50.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "fedbe2ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Confidence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feature ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000262225</th>\n",
       "      <td>d__Bacteria; p__Fusobacteriota; c__Fusobacteri...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G006716645</th>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G001310255</th>\n",
       "      <td>d__Bacteria; p__Proteobacteria; c__Alphaproteo...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G900604515</th>\n",
       "      <td>d__Bacteria; p__Actinobacteriota; c__Actinomyc...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G001813085</th>\n",
       "      <td>d__Bacteria; p__Actinobacteriota; c__Actinomyc...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        Taxon  Confidence\n",
       "Feature ID                                                               \n",
       "G000262225  d__Bacteria; p__Fusobacteriota; c__Fusobacteri...         1.0\n",
       "G006716645  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...         1.0\n",
       "G001310255  d__Bacteria; p__Proteobacteria; c__Alphaproteo...         1.0\n",
       "G900604515  d__Bacteria; p__Actinobacteriota; c__Actinomyc...         1.0\n",
       "G001813085  d__Bacteria; p__Actinobacteriota; c__Actinomyc...         1.0"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxonomy = pd.read_csv('../../U19_3XTG/data/Microbiome/pooling_filtered/gg2_taxonomy/taxonomy.tsv',\n",
    "                      sep='\\t', index_col=0)\n",
    "taxonomy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "75f02637",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Confidence</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feature ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000262225</th>\n",
       "      <td>d__Bacteria; p__Fusobacteriota; c__Fusobacteri...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Fusobacteriota</td>\n",
       "      <td>c__Fusobacteriia</td>\n",
       "      <td>o__Fusobacteriales_993521</td>\n",
       "      <td>f__Fusobacteriaceae_993521</td>\n",
       "      <td>g__Fusobacterium_C</td>\n",
       "      <td>s__Fusobacterium_C necrophorum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G006716645</th>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Sphingobacteriales</td>\n",
       "      <td>f__Sphingobacteriaceae</td>\n",
       "      <td>g__Pararcticibacter</td>\n",
       "      <td>s__Pararcticibacter tournemirensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G001310255</th>\n",
       "      <td>d__Bacteria; p__Proteobacteria; c__Alphaproteo...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Alphaproteobacteria</td>\n",
       "      <td>o__Caulobacterales</td>\n",
       "      <td>f__Caulobacteraceae</td>\n",
       "      <td>g__Brevundimonas</td>\n",
       "      <td>s__Brevundimonas aurantiaca</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G900604515</th>\n",
       "      <td>d__Bacteria; p__Actinobacteriota; c__Actinomyc...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Actinobacteriota</td>\n",
       "      <td>c__Actinomycetia</td>\n",
       "      <td>o__Actinomycetales</td>\n",
       "      <td>f__Actinomycetaceae</td>\n",
       "      <td>g__Arcanobacterium_A_386370</td>\n",
       "      <td>s__Arcanobacterium_A_386370 ihumii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G001813085</th>\n",
       "      <td>d__Bacteria; p__Actinobacteriota; c__Actinomyc...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Actinobacteriota</td>\n",
       "      <td>c__Actinomycetia</td>\n",
       "      <td>o__Actinomycetales</td>\n",
       "      <td>f__Actinomycetaceae</td>\n",
       "      <td>g__Gleimia</td>\n",
       "      <td>s__Gleimia europaea_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        Taxon  Confidence  \\\n",
       "Feature ID                                                                  \n",
       "G000262225  d__Bacteria; p__Fusobacteriota; c__Fusobacteri...         1.0   \n",
       "G006716645  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...         1.0   \n",
       "G001310255  d__Bacteria; p__Proteobacteria; c__Alphaproteo...         1.0   \n",
       "G900604515  d__Bacteria; p__Actinobacteriota; c__Actinomyc...         1.0   \n",
       "G001813085  d__Bacteria; p__Actinobacteriota; c__Actinomyc...         1.0   \n",
       "\n",
       "                 Domain                Phylum                    Class  \\\n",
       "Feature ID                                                               \n",
       "G000262225  d__Bacteria     p__Fusobacteriota         c__Fusobacteriia   \n",
       "G006716645  d__Bacteria       p__Bacteroidota           c__Bacteroidia   \n",
       "G001310255  d__Bacteria     p__Proteobacteria   c__Alphaproteobacteria   \n",
       "G900604515  d__Bacteria   p__Actinobacteriota         c__Actinomycetia   \n",
       "G001813085  d__Bacteria   p__Actinobacteriota         c__Actinomycetia   \n",
       "\n",
       "                                 Order                       Family  \\\n",
       "Feature ID                                                            \n",
       "G000262225   o__Fusobacteriales_993521   f__Fusobacteriaceae_993521   \n",
       "G006716645       o__Sphingobacteriales       f__Sphingobacteriaceae   \n",
       "G001310255          o__Caulobacterales          f__Caulobacteraceae   \n",
       "G900604515          o__Actinomycetales          f__Actinomycetaceae   \n",
       "G001813085          o__Actinomycetales          f__Actinomycetaceae   \n",
       "\n",
       "                                   Genus                              Species  \n",
       "Feature ID                                                                     \n",
       "G000262225            g__Fusobacterium_C       s__Fusobacterium_C necrophorum  \n",
       "G006716645           g__Pararcticibacter   s__Pararcticibacter tournemirensis  \n",
       "G001310255              g__Brevundimonas          s__Brevundimonas aurantiaca  \n",
       "G900604515   g__Arcanobacterium_A_386370   s__Arcanobacterium_A_386370 ihumii  \n",
       "G001813085                    g__Gleimia                s__Gleimia europaea_A  "
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "txs = ['Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species']\n",
    "taxonomy[txs] = taxonomy['Taxon'].str.split(';', expand=True)\n",
    "taxonomy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "d0759b26",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Phylum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005825-532</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>532</td>\n",
       "      <td>0.035114</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...</td>\n",
       "      <td>f__Bacillaceae_D_361077</td>\n",
       "      <td>g__Bacillus_S</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-703</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>703</td>\n",
       "      <td>0.360000</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...</td>\n",
       "      <td>f__Bacillaceae_D_361077</td>\n",
       "      <td>g__Bacillus_S</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-816</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>816</td>\n",
       "      <td>-0.007822</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...</td>\n",
       "      <td>f__Bacillaceae_D_361077</td>\n",
       "      <td>g__Bacillus_S</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-881</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>881</td>\n",
       "      <td>0.197130</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...</td>\n",
       "      <td>f__Bacillaceae_D_361077</td>\n",
       "      <td>g__Bacillus_S</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000005825-1017</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>1017</td>\n",
       "      <td>0.345076</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...</td>\n",
       "      <td>f__Bacillaceae_D_361077</td>\n",
       "      <td>g__Bacillus_S</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        OGU  metab_row_id  jointRPCA_co-occur_value  \\\n",
       "node                                                                  \n",
       "G000005825-532   G000005825           532                  0.035114   \n",
       "G000005825-703   G000005825           703                  0.360000   \n",
       "G000005825-816   G000005825           816                 -0.007822   \n",
       "G000005825-881   G000005825           881                  0.197130   \n",
       "G000005825-1017  G000005825          1017                  0.345076   \n",
       "\n",
       "                                                             Taxon  \\\n",
       "node                                                                 \n",
       "G000005825-532   d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...   \n",
       "G000005825-703   d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...   \n",
       "G000005825-816   d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...   \n",
       "G000005825-881   d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...   \n",
       "G000005825-1017  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...   \n",
       "\n",
       "                                   Family           Genus            Phylum  \n",
       "node                                                                         \n",
       "G000005825-532    f__Bacillaceae_D_361077   g__Bacillus_S   p__Firmicutes_D  \n",
       "G000005825-703    f__Bacillaceae_D_361077   g__Bacillus_S   p__Firmicutes_D  \n",
       "G000005825-816    f__Bacillaceae_D_361077   g__Bacillus_S   p__Firmicutes_D  \n",
       "G000005825-881    f__Bacillaceae_D_361077   g__Bacillus_S   p__Firmicutes_D  \n",
       "G000005825-1017   f__Bacillaceae_D_361077   g__Bacillus_S   p__Firmicutes_D  "
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog1_fecal = pd.merge(cyscp_3xtg_fecal, taxonomy[['Taxon', 'Family', 'Genus', 'Phylum']], \n",
    "                      left_on='OGU', right_index=True)\n",
    "tog1_brain = pd.merge(cyscp_3xtg_brain, taxonomy[['Taxon', 'Family', 'Genus', 'Phylum']], \n",
    "                      left_on='OGU', right_index=True)\n",
    "tog1_brain.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "fddb6910",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VIP1</th>\n",
       "      <th>SPF_genotype</th>\n",
       "      <th>VIP2</th>\n",
       "      <th>GF_genotype</th>\n",
       "      <th>VIP3</th>\n",
       "      <th>3xTG_colonization</th>\n",
       "      <th>VIP4</th>\n",
       "      <th>WT_colonization</th>\n",
       "      <th>VIP_mean</th>\n",
       "      <th>Exclusivity</th>\n",
       "      <th>...</th>\n",
       "      <th>Microbiome</th>\n",
       "      <th>mz</th>\n",
       "      <th>RT</th>\n",
       "      <th>Corr_ID</th>\n",
       "      <th>Compound_Name</th>\n",
       "      <th>Adduct_GNPS</th>\n",
       "      <th>Adduct_SIRIUS</th>\n",
       "      <th>NPC#pathway</th>\n",
       "      <th>NPC#superclass</th>\n",
       "      <th>NPC#class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feature_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>4.567</td>\n",
       "      <td>Mut</td>\n",
       "      <td>4.439</td>\n",
       "      <td>Mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.976</td>\n",
       "      <td>GF</td>\n",
       "      <td>3.661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>284.0986</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GUANOSINE - 20.0 eV</td>\n",
       "      <td>M+H</td>\n",
       "      <td>[M + H]+</td>\n",
       "      <td>Carbohydrates</td>\n",
       "      <td>Nucleosides</td>\n",
       "      <td>Purine nucleos(t)ides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>3.082</td>\n",
       "      <td>Mut</td>\n",
       "      <td>3.375</td>\n",
       "      <td>Mut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>269.0877</td>\n",
       "      <td>0.3721</td>\n",
       "      <td>23.0</td>\n",
       "      <td>INOSINE - 20.0 eV</td>\n",
       "      <td>M+H</td>\n",
       "      <td>[M + H]+</td>\n",
       "      <td>Carbohydrates</td>\n",
       "      <td>Nucleosides</td>\n",
       "      <td>Purine nucleos(t)ides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1417</th>\n",
       "      <td>2.784</td>\n",
       "      <td>WT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.523</td>\n",
       "      <td>GF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>266.1383</td>\n",
       "      <td>2.0846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[M + H]+</td>\n",
       "      <td>Fatty acids</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>2.430</td>\n",
       "      <td>WT</td>\n",
       "      <td>4.460</td>\n",
       "      <td>WT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.870</td>\n",
       "      <td>GF</td>\n",
       "      <td>2.920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>276.1439</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[M + H]+</td>\n",
       "      <td>Fatty acids</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>1.863</td>\n",
       "      <td>WT</td>\n",
       "      <td>4.053</td>\n",
       "      <td>WT</td>\n",
       "      <td>2.456</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>185.1284</td>\n",
       "      <td>0.7642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[M + H]+</td>\n",
       "      <td>Alkaloids</td>\n",
       "      <td>Lysine alkaloids</td>\n",
       "      <td>Piperidine alkaloids</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             VIP1 SPF_genotype   VIP2 GF_genotype   VIP3 3xTG_colonization  \\\n",
       "Feature_ID                                                                   \n",
       "383         4.567          Mut  4.439         Mut    NaN               NaN   \n",
       "600         3.082          Mut  3.375         Mut    NaN               NaN   \n",
       "1417        2.784           WT    NaN         NaN  3.523                GF   \n",
       "403         2.430           WT  4.460          WT    NaN               NaN   \n",
       "738         1.863           WT  4.053          WT  2.456               SPF   \n",
       "\n",
       "             VIP4 WT_colonization  VIP_mean Exclusivity  ... Microbiome  \\\n",
       "Feature_ID                                               ...              \n",
       "383         1.976              GF     3.661         NaN  ...         No   \n",
       "600           NaN             NaN     3.229         NaN  ...         No   \n",
       "1417          NaN             NaN     3.154         NaN  ...         No   \n",
       "403         1.870              GF     2.920         NaN  ...         No   \n",
       "738           NaN             NaN     2.791         NaN  ...         No   \n",
       "\n",
       "                  mz      RT  Corr_ID        Compound_Name Adduct_GNPS  \\\n",
       "Feature_ID                                                               \n",
       "383         284.0986  0.3671      NaN  GUANOSINE - 20.0 eV         M+H   \n",
       "600         269.0877  0.3721     23.0    INOSINE - 20.0 eV         M+H   \n",
       "1417        266.1383  2.0846      NaN                  NaN         NaN   \n",
       "403         276.1439  0.3963      NaN                  NaN         NaN   \n",
       "738         185.1284  0.7642      NaN                  NaN         NaN   \n",
       "\n",
       "           Adduct_SIRIUS    NPC#pathway    NPC#superclass  \\\n",
       "Feature_ID                                                  \n",
       "383             [M + H]+  Carbohydrates       Nucleosides   \n",
       "600             [M + H]+  Carbohydrates       Nucleosides   \n",
       "1417            [M + H]+    Fatty acids      Fatty esters   \n",
       "403             [M + H]+    Fatty acids      Fatty esters   \n",
       "738             [M + H]+      Alkaloids  Lysine alkaloids   \n",
       "\n",
       "                        NPC#class  \n",
       "Feature_ID                         \n",
       "383         Purine nucleos(t)ides  \n",
       "600         Purine nucleos(t)ides  \n",
       "1417        Fatty acyl carnitines  \n",
       "403         Fatty acyl carnitines  \n",
       "738          Piperidine alkaloids  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metab_id_fecal = pd.read_csv('../data/Metabolome/26112024/annotation/3xtg_fecal_spf_features_info_mn.csv',\n",
    "                            index_col=0)\n",
    "metab_id_brain = pd.read_csv('../data/Metabolome/26112024/annotation/3xtg_brain_features_info_mn.csv',\n",
    "                            index_col=0)\n",
    "metab_id_brain.head()\n",
    "#plan to use Compound_Name and NPC#class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b8c7bd72",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metab fecal  (2055, 17)\n",
      "metab brain  (268, 21)\n"
     ]
    }
   ],
   "source": [
    "print('metab fecal ', metab_id_fecal.shape)\n",
    "print('metab brain ', metab_id_brain.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "db859dd8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Compound_Name</th>\n",
       "      <th>NPC#class</th>\n",
       "      <th>NPC#superclass</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005825-221</th>\n",
       "      <td>G000005825</td>\n",
       "      <td>221</td>\n",
       "      <td>-0.012077</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...</td>\n",
       "      <td>f__Bacillaceae_D_361077</td>\n",
       "      <td>g__Bacillus_S</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>Small peptides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000006785-221</th>\n",
       "      <td>G000006785</td>\n",
       "      <td>221</td>\n",
       "      <td>-0.206482</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...</td>\n",
       "      <td>f__Streptococcaceae</td>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>Small peptides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000006865-221</th>\n",
       "      <td>G000006865</td>\n",
       "      <td>221</td>\n",
       "      <td>-0.006841</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...</td>\n",
       "      <td>f__Streptococcaceae</td>\n",
       "      <td>g__Lactococcus_A_346120</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>Small peptides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000006925-221</th>\n",
       "      <td>G000006925</td>\n",
       "      <td>221</td>\n",
       "      <td>-0.011491</td>\n",
       "      <td>d__Bacteria; p__Proteobacteria; c__Gammaproteo...</td>\n",
       "      <td>f__Enterobacteriaceae_A</td>\n",
       "      <td>g__Escherichia_710834</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>Small peptides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000007265-221</th>\n",
       "      <td>G000007265</td>\n",
       "      <td>221</td>\n",
       "      <td>-0.073954</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...</td>\n",
       "      <td>f__Streptococcaceae</td>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>Small peptides</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       OGU  metab_row_id  jointRPCA_co-occur_value  \\\n",
       "node                                                                 \n",
       "G000005825-221  G000005825           221                 -0.012077   \n",
       "G000006785-221  G000006785           221                 -0.206482   \n",
       "G000006865-221  G000006865           221                 -0.006841   \n",
       "G000006925-221  G000006925           221                 -0.011491   \n",
       "G000007265-221  G000007265           221                 -0.073954   \n",
       "\n",
       "                                                            Taxon  \\\n",
       "node                                                                \n",
       "G000005825-221  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__B...   \n",
       "G000006785-221  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...   \n",
       "G000006865-221  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...   \n",
       "G000006925-221  d__Bacteria; p__Proteobacteria; c__Gammaproteo...   \n",
       "G000007265-221  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...   \n",
       "\n",
       "                                  Family                     Genus  \\\n",
       "node                                                                 \n",
       "G000005825-221   f__Bacillaceae_D_361077             g__Bacillus_S   \n",
       "G000006785-221       f__Streptococcaceae          g__Streptococcus   \n",
       "G000006865-221       f__Streptococcaceae   g__Lactococcus_A_346120   \n",
       "G000006925-221   f__Enterobacteriaceae_A     g__Escherichia_710834   \n",
       "G000007265-221       f__Streptococcaceae          g__Streptococcus   \n",
       "\n",
       "                            Phylum Compound_Name   NPC#class  NPC#superclass  \n",
       "node                                                                          \n",
       "G000005825-221     p__Firmicutes_D           NaN  Aminoacids  Small peptides  \n",
       "G000006785-221     p__Firmicutes_D           NaN  Aminoacids  Small peptides  \n",
       "G000006865-221     p__Firmicutes_D           NaN  Aminoacids  Small peptides  \n",
       "G000006925-221   p__Proteobacteria           NaN  Aminoacids  Small peptides  \n",
       "G000007265-221     p__Firmicutes_D           NaN  Aminoacids  Small peptides  "
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog2_fecal = pd.merge(tog1_fecal, metab_id_fecal[['Compound_Name', 'NPC#class', 'NPC#superclass']], \n",
    "                      left_on='metab_row_id', right_index=True)\n",
    "tog2_brain = pd.merge(tog1_brain, metab_id_brain[['Compound_Name', 'NPC#class', 'NPC#superclass']], \n",
    "                      left_on='metab_row_id', right_index=True)\n",
    "tog2_brain.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "cdd5c5ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog2_fecal['metab_row_id'] = 'metab'+tog2_fecal['metab_row_id'].astype(str)\n",
    "tog2_brain['metab_row_id'] = 'metab'+tog2_brain['metab_row_id'].astype(str)\n",
    "#no underscores or dashes for formatting purposes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "c693072a",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog2_fecal['jRPCA_values_scaled'] = (tog2_fecal['jointRPCA_co-occur_value'] + 1) * 50\n",
    "tog2_brain['jRPCA_values_scaled'] = (tog2_brain['jointRPCA_co-occur_value'] + 1) * 50\n",
    "#does not work with negative values - convert scale from -1:1 to 0:100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "9a69500b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fecal  (3480416, 11)\n",
      "brain  (477402, 11)\n"
     ]
    }
   ],
   "source": [
    "print('fecal ', tog2_fecal.shape)\n",
    "print('brain ', tog2_brain.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "84010741",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog2_fecal.to_csv('../data/JointRPCA/3xtg_sac_fecal_cytoscape_format3.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "aa9dff5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog2_brain.to_csv('../data/JointRPCA/3xtg_brain_cytoscape_format3.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "0ce95877",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['OGU', 'metab_row_id', 'jointRPCA_co-occur_value', 'Taxon', 'Family',\n",
       "       'Genus', 'Phylum', 'Compound_Name', 'NPC#class', 'NPC#superclass',\n",
       "       'jRPCA_values_scaled'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog2_brain.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "0adab67a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def select_rows_with_multiple_values(df):\n",
    "    \"\"\"\n",
    "    Selects rows where the unique strings in 'OGU' or 'metab_row_id' have \n",
    "    multiple (more than 1) values in 'jRPCA_values_scaled' column \n",
    "    within the ranges 0-10 or 90-100.\n",
    "\n",
    "    Args:\n",
    "        df (pd.DataFrame): The input DataFrame.\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The filtered DataFrame.\n",
    "    \"\"\"\n",
    "\n",
    "    filtered_df = df[(df['jRPCA_values_scaled'] <= 10)|(90 <= df['jRPCA_values_scaled'])]\n",
    "    \n",
    "    group_counts1 = filtered_df.groupby('metab_row_id')['jRPCA_values_scaled'].count()\n",
    "    metab_selected_ids = group_counts1[group_counts1 > 100].index.tolist()\n",
    "    \n",
    "    group_counts2 = filtered_df.groupby('OGU')['jRPCA_values_scaled'].count()\n",
    "    micro_selected_ids = group_counts2[group_counts2 > 50].index.tolist()\n",
    "    \n",
    "    filtered_df2 = filtered_df[filtered_df['metab_row_id'].isin(metab_selected_ids)]\n",
    "    filtered_df3 = filtered_df2[filtered_df2['OGU'].isin(micro_selected_ids)]\n",
    "    \n",
    "    return filtered_df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "0293ba58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(100970, 10)"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog2_brainr = tog2_brain[(tog2_brain['jRPCA_values_scaled'] <= 10)|(90 <= tog2_brain['jRPCA_values_scaled'])]\n",
    "tog2_brainr.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "90fd3113",
   "metadata": {},
   "outputs": [],
   "source": [
    "group_counts1 = tog2_brainr.groupby('metab_row_id')['jRPCA_values_scaled'].count()\n",
    "metab_selected_ids = group_counts1[group_counts1 > 100].index.tolist()\n",
    "    \n",
    "group_counts2 = tog2_brainr.groupby('OGU')['jRPCA_values_scaled'].count()\n",
    "micro_selected_ids = group_counts2[group_counts2 > 50].index.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "fdc87f6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OGU</th>\n",
       "      <th>metab_row_id</th>\n",
       "      <th>jointRPCA_co-occur_value</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Compound_Name</th>\n",
       "      <th>NPC#class</th>\n",
       "      <th>NPC#superclass</th>\n",
       "      <th>jRPCA_values_scaled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000621705-357</th>\n",
       "      <td>G000621705</td>\n",
       "      <td>metab357</td>\n",
       "      <td>-0.895334</td>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "      <td>f__Prolixibacteraceae</td>\n",
       "      <td>g__Prolixibacter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>5.233308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000974365-357</th>\n",
       "      <td>G000974365</td>\n",
       "      <td>metab357</td>\n",
       "      <td>-0.890002</td>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "      <td>f__Marinilabiliaceae</td>\n",
       "      <td>g__Geofilum_869386</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>5.499898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G001483965-357</th>\n",
       "      <td>G001483965</td>\n",
       "      <td>metab357</td>\n",
       "      <td>-0.819508</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...</td>\n",
       "      <td>f__Carnobacteriaceae</td>\n",
       "      <td>g__Carnobacterium_A_320743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>9.024613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G002000245-357</th>\n",
       "      <td>G002000245</td>\n",
       "      <td>metab357</td>\n",
       "      <td>-0.944936</td>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "      <td>f__Sphingobacteriaceae</td>\n",
       "      <td>g__Sphingobacterium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>2.753190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G003234935-357</th>\n",
       "      <td>G003234935</td>\n",
       "      <td>metab357</td>\n",
       "      <td>-0.800186</td>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "      <td>f__Crocinitomicaceae</td>\n",
       "      <td>g__Putridiphycobacter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>9.990718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       OGU metab_row_id  jointRPCA_co-occur_value  \\\n",
       "node                                                                \n",
       "G000621705-357  G000621705     metab357                 -0.895334   \n",
       "G000974365-357  G000974365     metab357                 -0.890002   \n",
       "G001483965-357  G001483965     metab357                 -0.819508   \n",
       "G002000245-357  G002000245     metab357                 -0.944936   \n",
       "G003234935-357  G003234935     metab357                 -0.800186   \n",
       "\n",
       "                                                            Taxon  \\\n",
       "node                                                                \n",
       "G000621705-357  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...   \n",
       "G000974365-357  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...   \n",
       "G001483965-357  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__L...   \n",
       "G002000245-357  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...   \n",
       "G003234935-357  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...   \n",
       "\n",
       "                                 Family                        Genus  \\\n",
       "node                                                                   \n",
       "G000621705-357    f__Prolixibacteraceae             g__Prolixibacter   \n",
       "G000974365-357     f__Marinilabiliaceae           g__Geofilum_869386   \n",
       "G001483965-357     f__Carnobacteriaceae   g__Carnobacterium_A_320743   \n",
       "G002000245-357   f__Sphingobacteriaceae          g__Sphingobacterium   \n",
       "G003234935-357     f__Crocinitomicaceae        g__Putridiphycobacter   \n",
       "\n",
       "               Compound_Name               NPC#class     NPC#superclass  \\\n",
       "node                                                                      \n",
       "G000621705-357           NaN  Simple amide alkaloids  Peptide alkaloids   \n",
       "G000974365-357           NaN  Simple amide alkaloids  Peptide alkaloids   \n",
       "G001483965-357           NaN  Simple amide alkaloids  Peptide alkaloids   \n",
       "G002000245-357           NaN  Simple amide alkaloids  Peptide alkaloids   \n",
       "G003234935-357           NaN  Simple amide alkaloids  Peptide alkaloids   \n",
       "\n",
       "                jRPCA_values_scaled  \n",
       "node                                 \n",
       "G000621705-357             5.233308  \n",
       "G000974365-357             5.499898  \n",
       "G001483965-357             9.024613  \n",
       "G002000245-357             2.753190  \n",
       "G003234935-357             9.990718  "
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog2_brainr2 = tog2_brainr[tog2_brainr['metab_row_id'].isin(metab_selected_ids)]\n",
    "tog2_brainr3 = tog2_brainr2[tog2_brainr2['OGU'].isin(micro_selected_ids)]\n",
    "tog2_brainr3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "198b0d35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(62497, 10)"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog2_brainr3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "39c02ff9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(62497, 11)"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog3_brain = select_rows_with_multiple_values(tog2_brain)\n",
    "tog3_brain.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "05a5498c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1076752, 9)"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog3_fecal = select_rows_with_multiple_values(tog2_fecal)\n",
    "tog3_fecal.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "1a8334ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "import networkx as nx\n",
    "from sklearn.cluster import KMeans \n",
    "from pygraphml import Graph, Node, Edge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "0ef1c9e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_graphml_from_df(df, output_file_path, n_clusters, percentile_cutoff):\n",
    "    \"\"\"\n",
    "    Converts a jointRPCA correlation df to a GraphML file.\n",
    "\n",
    "    Args:\n",
    "        co-occurence table: A 2D NumPy array representing the distance matrix.\n",
    "        output_file: The name of the output GraphML file. Defaults to \"graph.graphml\".\n",
    "    \"\"\"\n",
    "\n",
    "    if df.shape[1] != 5:\n",
    "        raise ValueError(\"Table must have exactly 5 columns: microbe_id, microbe_name, metabolite_id, metabolite_name, correlation.\")\n",
    "\n",
    "    df.columns = ['microbe_id', 'metabolite_id', 'microbe_name', 'metabolite_name', 'correlation']\n",
    "\n",
    "    # Calculate correlation thresholds - no longer needed, filtered above\n",
    "    #lower_threshold = df['correlation'].quantile((100 - percentile_cutoff) / 100)\n",
    "    #upper_threshold = df['correlation'].quantile(percentile_cutoff / 100)\n",
    "\n",
    "    # Filter DataFrame based on correlation thresholds  - no longer needed, filtered above\n",
    "    #filtered_df = df[(df['correlation'] <= lower_threshold) | (df['correlation'] >= upper_threshold)]\n",
    "    filtered_df = df #just so I don't have to alter the rest of the code\n",
    "    \n",
    "    #Create a list for edge colors\n",
    "    edge_color = []\n",
    "    \n",
    "    # Create a graph\n",
    "    G = nx.Graph()\n",
    "\n",
    "    # Create lists of unique microbe and metabolite IDs\n",
    "    microbe_ids = filtered_df['microbe_id'].unique()\n",
    "    metabolite_ids = filtered_df['metabolite_id'].unique()\n",
    "\n",
    "    # Create feature matrices for clustering\n",
    "    microbe_feature_matrix = []\n",
    "    for cid in microbe_ids:\n",
    "        row = []\n",
    "        for tid in metabolite_ids:\n",
    "            # Get correlation value (or other relevant feature)\n",
    "            correlation = filtered_df[(filtered_df['microbe_id'] == cid) & (filtered_df['metabolite_id'] == tid)]['correlation'].values\n",
    "            row.append(correlation[0] if len(correlation) > 0 else 0)  # Handle missing correlations\n",
    "        microbe_feature_matrix.append(row)\n",
    "\n",
    "    metabolite_feature_matrix = []\n",
    "    for tid in metabolite_ids:\n",
    "        row = []\n",
    "        for cid in microbe_ids:\n",
    "            # Get correlation value (or other relevant feature)\n",
    "            correlation = filtered_df[(filtered_df['microbe_id'] == cid) & (filtered_df['metabolite_id'] == tid)]['correlation'].values\n",
    "            row.append(correlation[0] if len(correlation) > 0 else 0)  # Handle missing correlations\n",
    "        metabolite_feature_matrix.append(row)\n",
    "\n",
    "    # Perform k-means clustering\n",
    "    kmeans_microbe = KMeans(n_clusters=n_clusters, random_state=42)\n",
    "    microbe_clusters = kmeans_microbe.fit_predict(microbe_feature_matrix)\n",
    "\n",
    "    kmeans_metabolite = KMeans(n_clusters=n_clusters, random_state=42)\n",
    "    metabolite_clusters = kmeans_metabolite.fit_predict(metabolite_feature_matrix)\n",
    "    \n",
    "    \n",
    "    #Graph modifications\n",
    "    # Add nodes to the graph with cluster information\n",
    "    for index, row in filtered_df.iterrows():\n",
    "        microbe_id = row['microbe_id']\n",
    "        microbe_name = row['microbe_name']\n",
    "        metabolite_id = row['metabolite_id']\n",
    "        metabolite_name = row['metabolite_name']\n",
    "        correlation = row['correlation']\n",
    "        \n",
    "        # Get cluster assignments\n",
    "        microbe_cluster = microbe_clusters[list(microbe_ids).index(microbe_id)]\n",
    "        metabolite_cluster = metabolite_clusters[list(metabolite_ids).index(metabolite_id)]\n",
    "        \n",
    "        G.add_node(microbe_id, name=microbe_name, type=\"microbe\", cluster=microbe_cluster)\n",
    "        G.add_node(metabolite_id, name=metabolite_name, type=\"metabolite\", cluster=metabolite_cluster)\n",
    "        \n",
    "        # Create interaction string \n",
    "        interaction = f\"{microbe_id}_{metabolite_id}\"\n",
    "        \n",
    "        # Count the number of connections for each node\n",
    "        microbe_degrees = dict(G.degree(G.nodes()))\n",
    "        metabolite_degrees = dict(G.degree(G.nodes()))\n",
    "    \n",
    "        # Determine edge color   \n",
    "        if microbe_degrees[microbe_id] > 50:\n",
    "            edge_color = \"red\" # Change color for edges with multiple connections\n",
    "        elif metabolite_degrees[metabolite_id] > 100 :\n",
    "            edge_color = \"blue\"  # Change color for edges with multiple connections\n",
    "        else:\n",
    "            edge_color = \"black\"  # Default color\n",
    "        \n",
    "        G.add_edge(microbe_id, metabolite_id, weight=row['correlation'], \n",
    "                   interaction=interaction, color=edge_color)\n",
    "\n",
    "    nx.write_graphml(G, output_file_path)\n",
    "    print(f\"GraphML file created successfully: {output_file_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "80da1253",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog4_brain = tog3_brain[['OGU', 'metab_row_id', 'Genus', 'NPC#class', 'jRPCA_values_scaled']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "5df5cb8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "1fff1ae1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphML file created successfully: ../data/JointRPCA/3xtg_brain_microbe-metabolite_network.graphml\n",
      "total hours:  2.8404248519738515\n"
     ]
    }
   ],
   "source": [
    "t0 = time.time()\n",
    "create_graphml_from_df(tog4_brain, n_clusters=10, percentile_cutoff=75,\n",
    "                         output_file_path='../data/JointRPCA/3xtg_brain_microbe-metabolite_network.graphml')\n",
    "t1 = time.time()\n",
    "total = t1-t0\n",
    "print(\"total hours: \", total/3600)\n",
    "#generally has fewer than fecal\n",
    "#currently using NPC#class, but consider Compound_Name\n",
    "#took a little over 4 hours orginally, cutoff 75\n",
    "#with new modifications and extra features, now about 5 hours\n",
    "#with new filtering from above, no percentile cutoff, now about 3 hours"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e230c52",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "ab77fad0",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog5_brain = tog3_brain[['OGU', 'metab_row_id', 'Genus', 'Family', 'Phylum', 'NPC#class', 'NPC#superclass', 'jRPCA_values_scaled']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "2df4fe06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([' p__Bacteroidota', ' p__Firmicutes_D',\n",
       "       ' p__Desulfobacterota_G_459546', ' p__Firmicutes_A',\n",
       "       ' p__Proteobacteria', ' p__Desulfobacterota_I',\n",
       "       ' p__Fusobacteriota', ' p__Planctomycetota', ' p__Firmicutes_C',\n",
       "       ' p__Firmicutes_B_370529', ' p__Actinobacteriota',\n",
       "       ' p__Campylobacterota', ' p__Verrucomicrobiota',\n",
       "       ' p__Deferribacterota', ' p__Riflebacteria',\n",
       "       ' p__Bdellovibrionota_E', ' p__Patescibacteria',\n",
       "       ' p__Desulfobacterota_C', ' p__Cyanobacteria', ' p__Firmicutes_G'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog5_brain.Phylum.unique()\n",
    "#make these all levels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "74ff85b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Peptide alkaloids', nan, 'Tetramate alkaloids', 'Pseudoalkaloids',\n",
       "       'Small peptides', 'Nicotinic acid alkaloids', 'Fatty acyls',\n",
       "       'Ornithine alkaloids', 'Fatty esters', 'Saccharides',\n",
       "       'Aminosugars and aminoglycosides', 'Fatty Acids and Conjugates',\n",
       "       'Glycerophospholipids', 'Lysine alkaloids', 'Tyrosine alkaloids',\n",
       "       'Nucleosides', 'Eicosanoids', 'Fatty amides', 'Steroids',\n",
       "       'Tryptophan alkaloids', 'Linear polyketides', 'Oligopeptides',\n",
       "       'Histidine alkaloids', 'Phenolic acids (C6-C1)',\n",
       "       'Pseudoalkaloids (transamidation)'], dtype=object)"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog5_brain['NPC#superclass'].unique()\n",
    "#make these all levels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "b3e2250e",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog5_brain['collapsed_pair'] = tog5_brain['Genus'].astype(str)+\"-\"+tog5_brain['NPC#class']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "79267fe6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of all microbe-metabolite pairs: 62497\n",
      "Number of unique pairs: 11202\n"
     ]
    }
   ],
   "source": [
    "print(\"Number of all microbe-metabolite pairs:\", len(tog4_brain.collapsed_pair.values))\n",
    "print(\"Number of unique pairs:\", len(tog4_brain.collapsed_pair.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "4f088c00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count                                   44478\n",
       "unique                                  11201\n",
       "top        g__Limosilactobacillus-Tripeptides\n",
       "freq                                      276\n",
       "Name: collapsed_pair, dtype: object"
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog5_brain.collapsed_pair.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "e08d3ec7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(66, 1)"
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t5b = tog5_brain.collapsed_pair.value_counts().reset_index()\n",
    "t5b2 = t5b.set_index('index')\n",
    "t5b3 = t5b2[t5b2['collapsed_pair']>50]\n",
    "t5b3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "e6b494b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    11201.000000\n",
       "mean         3.970895\n",
       "std          9.400724\n",
       "min          1.000000\n",
       "25%          1.000000\n",
       "50%          2.000000\n",
       "75%          4.000000\n",
       "max        276.000000\n",
       "Name: collapsed_pair, dtype: float64"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t5b2['collapsed_pair'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "4f04efb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_graphml_from_dataframe(df, output_filename=\"network.graphml\", count_threshold=5):\n",
    "    \"\"\"\n",
    "    Creates a GraphML file from a Pandas DataFrame\n",
    "    Dataframe should be from jointRPCA co-occurence matrix\n",
    "    \n",
    "    Function allows visualizing microbe-metabolite relationships in Cytoscape.\n",
    "\n",
    "    Args:\n",
    "        df: Pandas DataFrame\n",
    "        output_filename: Name of the output GraphML file. defaults to network.graphml\n",
    "    \"\"\"\n",
    "    if df.shape[1] != 5:\n",
    "        raise ValueError(\"Table must have exactly 5 columns: microbe_id, microbe_name, metabolite_id, metabolite_name, correlation.\")\n",
    "\n",
    "    df.columns = ['microbe_name', 'metabolite_name', 'microbe_group', 'metabolite_group', 'correlation']    \n",
    "    \n",
    "    # Group and aggregate (optimized)\n",
    "    grouped = df.groupby(['microbe_name', 'metabolite_name', 'microbe_group', 'metabolite_group'])['correlation'].agg(['count', 'mean']).reset_index()\n",
    "\n",
    "    # Create NetworkX graph\n",
    "    graph = nx.Graph()\n",
    "\n",
    "    # Add the central node\n",
    "    graph.add_node(\"jointRPCA\", type=\"central\")  # Add type attribute\n",
    "\n",
    "    # Add nodes and edges for hierarchy\n",
    "    for _, row in df.iterrows():\n",
    "        microbe_phylum = row['microbe_group']\n",
    "        microbe_genus = row['microbe_name']\n",
    "        metabolite_superclass = row['metabolite_group']\n",
    "        metabolite_class = row['metabolite_name']\n",
    "\n",
    "        if not graph.has_node(microbe_phylum):\n",
    "            graph.add_node(microbe_phylum, type=\"phylum\")\n",
    "        if not graph.has_node(metabolite_superclass):\n",
    "            graph.add_node(metabolite_superclass, type=\"superclass\")\n",
    "        if not graph.has_node(microbe_genus):\n",
    "            graph.add_node(microbe_genus, type=\"genus\", count=0)\n",
    "        if not graph.has_node(metabolite_class):\n",
    "            graph.add_node(metabolite_class, type=\"class\", count=0)\n",
    "\n",
    "        graph.add_edge(\"jointRPCA\", microbe_phylum, type='jointRPCA-Phylum')\n",
    "        graph.add_edge(\"jointRPCA\", metabolite_superclass, type='jointRPCA-SuperClass')\n",
    "        graph.add_edge(microbe_phylum, microbe_genus, type='Phylum-Genus')\n",
    "        graph.add_edge(metabolite_superclass, metabolite_class, type='SuperClass-Class')\n",
    "\n",
    "\n",
    "    microbe_counts = grouped.groupby('microbe_name')['count'].sum().to_dict()\n",
    "    metabolite_counts = grouped.groupby('metabolite_name')['count'].sum().to_dict()\n",
    "\n",
    "    for _, row in grouped.iterrows():\n",
    "        microbe = row['microbe_name']\n",
    "        metabolite = row['metabolite_name']\n",
    "        correlation = row['mean']\n",
    "\n",
    "        count = microbe_counts.get(microbe, 0) + metabolite_counts.get(metabolite, 0)\n",
    "        if count > count_threshold:  # Check the threshold here!\n",
    "            total_count = microbe_counts.get(microbe, 0) + metabolite_counts.get(metabolite, 0)\n",
    "            if graph.has_node(microbe):\n",
    "                graph.nodes[microbe]['count'] = total_count\n",
    "            if graph.has_node(metabolite):\n",
    "                graph.nodes[metabolite]['count'] = total_count\n",
    "#       else: #If count is below the threshold, don't include the count attribute\n",
    "#         if graph.has_node(microbe):\n",
    "#             graph.nodes[microbe]['count'] = count\n",
    "#         if graph.has_node(metabolite):\n",
    "#             graph.nodes[metabolite]['count'] = count\n",
    "\n",
    "        graph.add_edge(microbe, metabolite, correlation=correlation, type=\"interaction\")\n",
    "\n",
    "        if correlation > 98:\n",
    "            graph[microbe][metabolite]['color'] = 'red'\n",
    "        elif correlation < 2:\n",
    "            graph[microbe][metabolite]['color'] = 'blue'\n",
    "        else:\n",
    "            graph[microbe][metabolite]['color'] = 'gray'\n",
    "\n",
    "    nx.write_graphml(graph, output_filename)  # Use networkx's write_graphml\n",
    "    print(f\"GraphML file created: {output_filename}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "90200460",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog6_brain = tog5_brain[['Genus', 'NPC#class', 'Phylum', 'NPC#superclass', 'jRPCA_values_scaled']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "234cfb99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphML file created: ../data/JointRPCA/3xtg_brain_microbe-metabolite_network2.graphml\n",
      "total hours:  0.0005200694004694621\n"
     ]
    }
   ],
   "source": [
    "t0 = time.time()\n",
    "create_graphml_from_dataframe(tog6_brain, count_threshold=50,\n",
    "                              output_filename='../data/JointRPCA/3xtg_brain_microbe-metabolite_network2.graphml')\n",
    "t1 = time.time()\n",
    "total = t1-t0\n",
    "print(\"total hours: \", total/3600)\n",
    "#This took a second"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e6a043a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f18128c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adb13eff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need the below file added on"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "6ec93ca7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>microbe_name</th>\n",
       "      <th>metabolite_name</th>\n",
       "      <th>microbe_group</th>\n",
       "      <th>metabolite_group</th>\n",
       "      <th>correlation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>node</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000621705-357</th>\n",
       "      <td>g__Prolixibacter</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>5.233308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000974365-357</th>\n",
       "      <td>g__Geofilum_869386</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>5.499898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G001483965-357</th>\n",
       "      <td>g__Carnobacterium_A_320743</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>9.024613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G002000245-357</th>\n",
       "      <td>g__Sphingobacterium</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>2.753190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G003234935-357</th>\n",
       "      <td>g__Putridiphycobacter</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>9.990718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               microbe_name         metabolite_name  \\\n",
       "node                                                                  \n",
       "G000621705-357             g__Prolixibacter  Simple amide alkaloids   \n",
       "G000974365-357           g__Geofilum_869386  Simple amide alkaloids   \n",
       "G001483965-357   g__Carnobacterium_A_320743  Simple amide alkaloids   \n",
       "G002000245-357          g__Sphingobacterium  Simple amide alkaloids   \n",
       "G003234935-357        g__Putridiphycobacter  Simple amide alkaloids   \n",
       "\n",
       "                   microbe_group   metabolite_group  correlation  \n",
       "node                                                              \n",
       "G000621705-357   p__Bacteroidota  Peptide alkaloids     5.233308  \n",
       "G000974365-357   p__Bacteroidota  Peptide alkaloids     5.499898  \n",
       "G001483965-357   p__Firmicutes_D  Peptide alkaloids     9.024613  \n",
       "G002000245-357   p__Bacteroidota  Peptide alkaloids     2.753190  \n",
       "G003234935-357   p__Bacteroidota  Peptide alkaloids     9.990718  "
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tog6_brain.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "5dc773dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node</th>\n",
       "      <th>microbe_name</th>\n",
       "      <th>metabolite_name</th>\n",
       "      <th>microbe_group</th>\n",
       "      <th>metabolite_group</th>\n",
       "      <th>correlation</th>\n",
       "      <th>microbe_id</th>\n",
       "      <th>metabolite_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G000621705-357</td>\n",
       "      <td>g__Prolixibacter</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>5.233308</td>\n",
       "      <td>G000621705</td>\n",
       "      <td>357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G000974365-357</td>\n",
       "      <td>g__Geofilum_869386</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>5.499898</td>\n",
       "      <td>G000974365</td>\n",
       "      <td>357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G001483965-357</td>\n",
       "      <td>g__Carnobacterium_A_320743</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>9.024613</td>\n",
       "      <td>G001483965</td>\n",
       "      <td>357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G002000245-357</td>\n",
       "      <td>g__Sphingobacterium</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>2.753190</td>\n",
       "      <td>G002000245</td>\n",
       "      <td>357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G003234935-357</td>\n",
       "      <td>g__Putridiphycobacter</td>\n",
       "      <td>Simple amide alkaloids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Peptide alkaloids</td>\n",
       "      <td>9.990718</td>\n",
       "      <td>G003234935</td>\n",
       "      <td>357</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             node                 microbe_name         metabolite_name  \\\n",
       "0  G000621705-357             g__Prolixibacter  Simple amide alkaloids   \n",
       "1  G000974365-357           g__Geofilum_869386  Simple amide alkaloids   \n",
       "2  G001483965-357   g__Carnobacterium_A_320743  Simple amide alkaloids   \n",
       "3  G002000245-357          g__Sphingobacterium  Simple amide alkaloids   \n",
       "4  G003234935-357        g__Putridiphycobacter  Simple amide alkaloids   \n",
       "\n",
       "      microbe_group   metabolite_group  correlation  microbe_id metabolite_id  \n",
       "0   p__Bacteroidota  Peptide alkaloids     5.233308  G000621705           357  \n",
       "1   p__Bacteroidota  Peptide alkaloids     5.499898  G000974365           357  \n",
       "2   p__Firmicutes_D  Peptide alkaloids     9.024613  G001483965           357  \n",
       "3   p__Bacteroidota  Peptide alkaloids     2.753190  G002000245           357  \n",
       "4   p__Bacteroidota  Peptide alkaloids     9.990718  G003234935           357  "
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_countz = tog6_brain.reset_index()\n",
    "brain_countz[['microbe_id', 'metabolite_id']] = brain_countz['node'].str.split(\"-\", expand=True)\n",
    "brain_countz.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "6ad82452",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>microbe_name</th>\n",
       "      <th>metabolite_name</th>\n",
       "      <th>microbe_group</th>\n",
       "      <th>metabolite_group</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g__</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g__</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>p__Desulfobacterota_G_459546</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g__</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g__</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>p__Verrucomicrobiota</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g__</td>\n",
       "      <td>Branched fatty acids</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Fatty Acids and Conjugates</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  microbe_name       metabolite_name                  microbe_group  \\\n",
       "0          g__            Aminoacids                p__Bacteroidota   \n",
       "1          g__            Aminoacids   p__Desulfobacterota_G_459546   \n",
       "2          g__            Aminoacids              p__Proteobacteria   \n",
       "3          g__            Aminoacids           p__Verrucomicrobiota   \n",
       "4          g__  Branched fatty acids                p__Bacteroidota   \n",
       "\n",
       "             metabolite_group  counts  \n",
       "0              Small peptides       8  \n",
       "1              Small peptides       3  \n",
       "2              Small peptides       6  \n",
       "3              Small peptides       5  \n",
       "4  Fatty Acids and Conjugates       1  "
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts = tog6_brain.groupby(['microbe_name', 'metabolite_name', \n",
    "                                   'microbe_group', 'metabolite_group'\n",
    "                                  ]).size().reset_index(name='counts')\n",
    "brain_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "5c3bff62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    13771.000000\n",
       "mean         3.229831\n",
       "std          7.962164\n",
       "min          1.000000\n",
       "25%          1.000000\n",
       "50%          1.000000\n",
       "75%          3.000000\n",
       "max        274.000000\n",
       "Name: counts, dtype: float64"
      ]
     },
     "execution_count": 288,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts.counts.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "6b9791ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13771, 5)"
      ]
     },
     "execution_count": 289,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "a67756d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(279, 5)"
      ]
     },
     "execution_count": 302,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts20 = brain_counts[brain_counts.counts>=20].copy()\n",
    "brain_counts20.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "5c59c330",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Tripeptides', 'Aminoacids', 'Dipeptides',\n",
       "       'Glycerophosphocholines', 'Linear peptides', 'Thia fatty acids',\n",
       "       'Lipopeptides', 'Fatty acyl carnitines', 'Purine nucleos(t)ides',\n",
       "       'Cyclic peptides', 'Gallotannins', 'Glycerophosphoethanolamines',\n",
       "       'Oxidized glycerophospholipids', 'Thiodiketopiperazine alkaloids',\n",
       "       'Pyridine alkaloids', 'Branched fatty acids', 'Cephalosporins',\n",
       "       'Imidazole alkaloids', 'N-acyl amines',\n",
       "       'N-acyl ethanolamines (endocannabinoids)', 'Polyamines',\n",
       "       'Pyrrolidine alkaloids', 'Simple amide alkaloids',\n",
       "       'Purine alkaloids', 'Wax monoesters'], dtype=object)"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts20.metabolite_name.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "20df1814",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Small peptides', 'Glycerophospholipids', 'Oligopeptides',\n",
       "       'Fatty Acids and Conjugates', 'Fatty esters', 'Nucleosides',\n",
       "       'Phenolic acids (C6-C1)', 'Nicotinic acid alkaloids',\n",
       "       'Tryptophan alkaloids', 'Linear polyketides', 'Lysine alkaloids',\n",
       "       'Histidine alkaloids', 'Fatty amides', 'Ornithine alkaloids',\n",
       "       'Aminosugars and aminoglycosides', 'Pseudoalkaloids'], dtype=object)"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts20.metabolite_group.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "7be6f8fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>microbe_name</th>\n",
       "      <th>metabolite_name</th>\n",
       "      <th>microbe_group</th>\n",
       "      <th>metabolite_group</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3610</th>\n",
       "      <td>g__Clostridium_T</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Firmicutes_A</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4023</th>\n",
       "      <td>g__Cryptobacteroides</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5450</th>\n",
       "      <td>g__Escherichia_710834</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8228</th>\n",
       "      <td>g__Ligilactobacillus</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8327</th>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8376</th>\n",
       "      <td>g__Liquorilactobacillus</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10184</th>\n",
       "      <td>g__Pediococcus</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12018</th>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>Fatty acyl carnitines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Fatty esters</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   microbe_name        metabolite_name       microbe_group  \\\n",
       "3610           g__Clostridium_T  Fatty acyl carnitines     p__Firmicutes_A   \n",
       "4023       g__Cryptobacteroides  Fatty acyl carnitines     p__Bacteroidota   \n",
       "5450      g__Escherichia_710834  Fatty acyl carnitines   p__Proteobacteria   \n",
       "8228       g__Ligilactobacillus  Fatty acyl carnitines     p__Firmicutes_D   \n",
       "8327     g__Limosilactobacillus  Fatty acyl carnitines     p__Firmicutes_D   \n",
       "8376    g__Liquorilactobacillus  Fatty acyl carnitines     p__Firmicutes_D   \n",
       "10184            g__Pediococcus  Fatty acyl carnitines     p__Firmicutes_D   \n",
       "12018          g__Streptococcus  Fatty acyl carnitines     p__Firmicutes_D   \n",
       "\n",
       "      metabolite_group  counts  \n",
       "3610      Fatty esters      29  \n",
       "4023      Fatty esters      31  \n",
       "5450      Fatty esters      31  \n",
       "8228      Fatty esters      34  \n",
       "8327      Fatty esters      37  \n",
       "8376      Fatty esters      30  \n",
       "10184     Fatty esters      22  \n",
       "12018     Fatty esters      20  "
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts20[brain_counts20.metabolite_name=='Fatty acyl carnitines']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "c655f10d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>microbe_name</th>\n",
       "      <th>metabolite_name</th>\n",
       "      <th>microbe_group</th>\n",
       "      <th>metabolite_group</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3643</th>\n",
       "      <td>g__Clostridium_T</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_A</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4055</th>\n",
       "      <td>g__Cryptobacteroides</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5449</th>\n",
       "      <td>g__Escherichia_710834</td>\n",
       "      <td>Dipeptides</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5483</th>\n",
       "      <td>g__Escherichia_710834</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7750</th>\n",
       "      <td>g__Lacticaseibacillus</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7794</th>\n",
       "      <td>g__Lactobacillus</td>\n",
       "      <td>Dipeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7797</th>\n",
       "      <td>g__Lactobacillus</td>\n",
       "      <td>Glycerophosphocholines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Glycerophospholipids</td>\n",
       "      <td>160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7817</th>\n",
       "      <td>g__Lactobacillus</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8259</th>\n",
       "      <td>g__Ligilactobacillus</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8316</th>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8326</th>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>Dipeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8329</th>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>Glycerophosphocholines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Glycerophospholipids</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8363</th>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8405</th>\n",
       "      <td>g__Liquorilactobacillus</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12007</th>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>Aminoacids</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12017</th>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>Dipeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12020</th>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>Glycerophosphocholines</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Glycerophospholipids</td>\n",
       "      <td>177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12051</th>\n",
       "      <td>g__Streptococcus</td>\n",
       "      <td>Tripeptides</td>\n",
       "      <td>p__Firmicutes_D</td>\n",
       "      <td>Small peptides</td>\n",
       "      <td>266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   microbe_name         metabolite_name       microbe_group  \\\n",
       "3643           g__Clostridium_T             Tripeptides     p__Firmicutes_A   \n",
       "4055       g__Cryptobacteroides             Tripeptides     p__Bacteroidota   \n",
       "5449      g__Escherichia_710834              Dipeptides   p__Proteobacteria   \n",
       "5483      g__Escherichia_710834             Tripeptides   p__Proteobacteria   \n",
       "7750      g__Lacticaseibacillus             Tripeptides     p__Firmicutes_D   \n",
       "7794           g__Lactobacillus              Dipeptides     p__Firmicutes_D   \n",
       "7797           g__Lactobacillus  Glycerophosphocholines     p__Firmicutes_D   \n",
       "7817           g__Lactobacillus             Tripeptides     p__Firmicutes_D   \n",
       "8259       g__Ligilactobacillus             Tripeptides     p__Firmicutes_D   \n",
       "8316     g__Limosilactobacillus              Aminoacids     p__Firmicutes_D   \n",
       "8326     g__Limosilactobacillus              Dipeptides     p__Firmicutes_D   \n",
       "8329     g__Limosilactobacillus  Glycerophosphocholines     p__Firmicutes_D   \n",
       "8363     g__Limosilactobacillus             Tripeptides     p__Firmicutes_D   \n",
       "8405    g__Liquorilactobacillus             Tripeptides     p__Firmicutes_D   \n",
       "12007          g__Streptococcus              Aminoacids     p__Firmicutes_D   \n",
       "12017          g__Streptococcus              Dipeptides     p__Firmicutes_D   \n",
       "12020          g__Streptococcus  Glycerophosphocholines     p__Firmicutes_D   \n",
       "12051          g__Streptococcus             Tripeptides     p__Firmicutes_D   \n",
       "\n",
       "           metabolite_group  counts  \n",
       "3643         Small peptides     102  \n",
       "4055         Small peptides     123  \n",
       "5449         Small peptides     107  \n",
       "5483         Small peptides     223  \n",
       "7750         Small peptides     112  \n",
       "7794         Small peptides     117  \n",
       "7797   Glycerophospholipids     160  \n",
       "7817         Small peptides     201  \n",
       "8259         Small peptides     179  \n",
       "8316         Small peptides     113  \n",
       "8326         Small peptides     140  \n",
       "8329   Glycerophospholipids     118  \n",
       "8363         Small peptides     274  \n",
       "8405         Small peptides     135  \n",
       "12007        Small peptides     101  \n",
       "12017        Small peptides     139  \n",
       "12020  Glycerophospholipids     177  \n",
       "12051        Small peptides     266  "
      ]
     },
     "execution_count": 308,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_counts20[brain_counts20.counts>100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "7906ce84",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = brain_counts20\n",
    "for (mic_id, met_id, mic_group, met_group), group in df.groupby(['microbe_name', 'metabolite_name', 'microbe_group', 'metabolite_group']):\n",
    "        mic_group_counts = df[df['microbe_group'] == mic_group]['counts'].sum()\n",
    "        met_group_counts = df[df['metabolite_group'] == met_group]['counts'].sum()\n",
    "        mic_id_counts = df[(df['microbe_group'] == mic_group) & (df['microbe_name'] == mic_id)]['counts'].sum()\n",
    "        met_id_counts = df[(df['metabolite_group'] == met_group) & (df['metabolite_name'] == met_id)]['counts'].sum()\n",
    "        g_counts = group['counts'].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "940e7d94",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8160\n",
      "215\n",
      "7101\n",
      "4243\n",
      "83\n"
     ]
    }
   ],
   "source": [
    "print(mic_group_counts)\n",
    "print(mic_id_counts)\n",
    "print(met_group_counts)\n",
    "print(met_id_counts)\n",
    "print(g_counts)\n",
    "#counts the number of jointRPCA co-occurences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "id": "40fcb3ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export information to import into Cytoscape and connect to graphml\n",
    "\n",
    "def transform_data(df, df_orig, organ):\n",
    "    # Creating an empty list to store dfs\n",
    "    transformed_dfs = [] \n",
    "\n",
    "    # Aggregating at each level and calculating node counts\n",
    "    for (mic_id, met_id, mic_group, met_group), group in df.groupby(['microbe_name', 'metabolite_name', 'microbe_group', 'metabolite_group']):\n",
    "        mic_group_counts = df[df['microbe_group'] == mic_group]['counts'].sum()\n",
    "        met_group_counts = df[df['metabolite_group'] == met_group]['counts'].sum()\n",
    "        mic_id_counts = df[(df['microbe_group'] == mic_group) & (df['microbe_name'] == mic_id)]['counts'].sum()\n",
    "        met_id_counts = df[(df['metabolite_group'] == met_group) & (df['metabolite_name'] == met_id)]['counts'].sum()\n",
    "        pair_counts = group['counts'].sum()\n",
    "\n",
    "        # Define levels and their respective counts\n",
    "        levels = [\n",
    "            (mic_group, 'level_1', mic_group_counts),\n",
    "            (met_group, 'level_1', met_group_counts),\n",
    "            (f'{mic_group}_{mic_id}', 'level_2', mic_id_counts),\n",
    "            (f'{met_group}_{met_id}', 'level_2', met_id_counts),\n",
    "            (f'{mic_id}_{met_id}', 'level_3', pair_counts),\n",
    "        ]\n",
    "\n",
    "        # Adding entries for each level\n",
    "        for node, level, counts in levels:\n",
    "            node_count = counts\n",
    "            new_df = pd.DataFrame([{  # Create a small DataFrame for each row\n",
    "                'node': node,\n",
    "                'level': level,\n",
    "                'counts': counts,\n",
    "                'node_counts': f'{node} ({node_count})'\n",
    "            }])\n",
    "            transformed_dfs.append(new_df)  # Append to the list of DataFrames\n",
    "    \n",
    "    # Adding the organism row\n",
    "    total_rows = len(df_orig)  # Gets the number of rows in the input dataframe\n",
    "    organ_row = {'node': organ,\n",
    "                 'level': 'level_0',\n",
    "                 'counts': total_rows,\n",
    "                 'node_counts': f'{organ} ({total_rows})'}\n",
    "    \n",
    "    #create the df\n",
    "    organ_df = pd.DataFrame([organ_row])\n",
    "    \n",
    "    #add to df list\n",
    "    transformed_dfs.append(organ_df)\n",
    "    \n",
    "    #combine all dfs\n",
    "    transformed_df = pd.concat(transformed_dfs, ignore_index=True)\n",
    "\n",
    "    # Removing duplicate entries and resetting index\n",
    "    transformed_df = transformed_df.drop_duplicates().reset_index(drop=True)\n",
    "\n",
    "    # add another column label_counts\n",
    "    transformed_df['label_counts'] = transformed_df['node'].str.split('_').str[-1] + ' (' + transformed_df['counts'].astype(str) + ')'\n",
    "\n",
    "    return transformed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "90841f37",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_nodes = transform_data(brain_counts20, tog6_brain, organ='brain')\n",
    "df_nodes.to_csv('../data/JointRPCA/3xtg_brain_additional_info_Cytoscape.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "b961a001",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node</th>\n",
       "      <th>level</th>\n",
       "      <th>counts</th>\n",
       "      <th>node_counts</th>\n",
       "      <th>label_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>p__Firmicutes_A</td>\n",
       "      <td>level_1</td>\n",
       "      <td>627</td>\n",
       "      <td>p__Firmicutes_A (627)</td>\n",
       "      <td>A (627)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Small peptides</td>\n",
       "      <td>level_1</td>\n",
       "      <td>7101</td>\n",
       "      <td>Small peptides (7101)</td>\n",
       "      <td>Small peptides (7101)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>p__Firmicutes_A_ g__Acetivibrio</td>\n",
       "      <td>level_2</td>\n",
       "      <td>27</td>\n",
       "      <td>p__Firmicutes_A_ g__Acetivibrio (27)</td>\n",
       "      <td>Acetivibrio (27)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Small peptides_Tripeptides</td>\n",
       "      <td>level_2</td>\n",
       "      <td>4243</td>\n",
       "      <td>Small peptides_Tripeptides (4243)</td>\n",
       "      <td>Tripeptides (4243)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g__Acetivibrio_Tripeptides</td>\n",
       "      <td>level_3</td>\n",
       "      <td>27</td>\n",
       "      <td>g__Acetivibrio_Tripeptides (27)</td>\n",
       "      <td>Tripeptides (27)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>g__Weissella_A_338544_Dipeptides</td>\n",
       "      <td>level_3</td>\n",
       "      <td>42</td>\n",
       "      <td>g__Weissella_A_338544_Dipeptides (42)</td>\n",
       "      <td>Dipeptides (42)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>g__Weissella_A_338544_Glycerophosphocholines</td>\n",
       "      <td>level_3</td>\n",
       "      <td>36</td>\n",
       "      <td>g__Weissella_A_338544_Glycerophosphocholines ...</td>\n",
       "      <td>Glycerophosphocholines (36)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>g__Weissella_A_338544_Lipopeptides</td>\n",
       "      <td>level_3</td>\n",
       "      <td>20</td>\n",
       "      <td>g__Weissella_A_338544_Lipopeptides (20)</td>\n",
       "      <td>Lipopeptides (20)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>g__Weissella_A_338544_Tripeptides</td>\n",
       "      <td>level_3</td>\n",
       "      <td>83</td>\n",
       "      <td>g__Weissella_A_338544_Tripeptides (83)</td>\n",
       "      <td>Tripeptides (83)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>brain</td>\n",
       "      <td>level_0</td>\n",
       "      <td>62497</td>\n",
       "      <td>brain (62497)</td>\n",
       "      <td>brain (62497)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>420 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              node    level  counts  \\\n",
       "0                                  p__Firmicutes_A  level_1     627   \n",
       "1                                   Small peptides  level_1    7101   \n",
       "2                  p__Firmicutes_A_ g__Acetivibrio  level_2      27   \n",
       "3                       Small peptides_Tripeptides  level_2    4243   \n",
       "4                       g__Acetivibrio_Tripeptides  level_3      27   \n",
       "..                                             ...      ...     ...   \n",
       "415               g__Weissella_A_338544_Dipeptides  level_3      42   \n",
       "416   g__Weissella_A_338544_Glycerophosphocholines  level_3      36   \n",
       "417             g__Weissella_A_338544_Lipopeptides  level_3      20   \n",
       "418              g__Weissella_A_338544_Tripeptides  level_3      83   \n",
       "419                                          brain  level_0   62497   \n",
       "\n",
       "                                           node_counts  \\\n",
       "0                                p__Firmicutes_A (627)   \n",
       "1                                Small peptides (7101)   \n",
       "2                 p__Firmicutes_A_ g__Acetivibrio (27)   \n",
       "3                    Small peptides_Tripeptides (4243)   \n",
       "4                      g__Acetivibrio_Tripeptides (27)   \n",
       "..                                                 ...   \n",
       "415              g__Weissella_A_338544_Dipeptides (42)   \n",
       "416   g__Weissella_A_338544_Glycerophosphocholines ...   \n",
       "417            g__Weissella_A_338544_Lipopeptides (20)   \n",
       "418             g__Weissella_A_338544_Tripeptides (83)   \n",
       "419                                      brain (62497)   \n",
       "\n",
       "                    label_counts  \n",
       "0                        A (627)  \n",
       "1          Small peptides (7101)  \n",
       "2               Acetivibrio (27)  \n",
       "3             Tripeptides (4243)  \n",
       "4               Tripeptides (27)  \n",
       "..                           ...  \n",
       "415              Dipeptides (42)  \n",
       "416  Glycerophosphocholines (36)  \n",
       "417            Lipopeptides (20)  \n",
       "418             Tripeptides (83)  \n",
       "419                brain (62497)  \n",
       "\n",
       "[420 rows x 5 columns]"
      ]
     },
     "execution_count": 346,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "id": "654c0b38",
   "metadata": {},
   "outputs": [],
   "source": [
    "def transform_data_sheet(df, focus, organ):\n",
    "    df_all = df.reset_index()\n",
    "    df_all[['microbe_id', 'metabolite_id']] = df_all['node'].str.split(\"-\", expand=True)\n",
    "    \n",
    "    # Creating an empty list to store dfs\n",
    "    transformed_dfs = [] \n",
    "\n",
    "    # Aggregating at each level and calculating node counts\n",
    "    if focus == \"metabolites\":\n",
    "        metab_df = df_all[['metabolite_group', 'metabolite_name', 'metabolite_id', 'correlation']].copy()\n",
    "        metab_df2 = metab_df.groupby(['metabolite_group', 'metabolite_name', \n",
    "                           'metabolite_id']).size().reset_index(name='counts')\n",
    "        for (met_group, met_class, met_id), group in metab_df2.groupby(['metabolite_group', 'metabolite_name', 'metabolite_id']):\n",
    "            met_group_counts = metab_df2[metab_df2['metabolite_group'] == met_group]['counts'].sum()\n",
    "            met_name_counts = metab_df2[(metab_df2['metabolite_group'] == met_group) & (metab_df2['metabolite_name'] == met_class)]['counts'].sum()\n",
    "            met_id_counts = group['counts'].sum()\n",
    "\n",
    "            # Define levels and their respective counts\n",
    "            levels = [\n",
    "                (met_group, 'level_1', met_group_counts),\n",
    "                (f'{met_group}_{met_class}', 'level_2', met_name_counts),\n",
    "                (f'{met_group}_{met_class}_{met_id}', 'level_3', met_id_counts),\n",
    "            ]\n",
    "\n",
    "            # Adding entries for each level\n",
    "            for node, level, counts in levels:\n",
    "                node_count = counts\n",
    "                new_df = pd.DataFrame([{  # Create a small DataFrame for each row\n",
    "                    'node': node,\n",
    "                    'level': level,\n",
    "                    'counts': counts,\n",
    "                    'node_counts': f'{node} ({node_count})'\n",
    "                }])\n",
    "                transformed_dfs.append(new_df)  # Append to the list of DataFrames\n",
    "    \n",
    "    elif focus == \"microbes\":\n",
    "        micro_df = df_all[['microbe_group', 'microbe_name', 'microbe_id', 'correlation']].copy()\n",
    "        micro_df2 = micro_df.groupby(['microbe_group', 'microbe_name', \n",
    "                           'microbe_id']).size().reset_index(name='counts')\n",
    "        \n",
    "        for (mic_group, mic_genus, mic_id), group in micro_df2.groupby(['microbe_group', 'microbe_name', 'microbe_id']):\n",
    "            mic_group_counts = micro_df2[micro_df2['microbe_group'] == mic_group]['counts'].sum()\n",
    "            mic_name_counts = micro_df2[(micro_df2['microbe_group'] == mic_group) & (micro_df2['microbe_name'] == mic_genus)]['counts'].sum()\n",
    "            mic_id_counts = group['counts'].sum()\n",
    "\n",
    "            # Define levels and their respective counts\n",
    "            levels = [\n",
    "                (mic_group, 'level_1', mic_group_counts),\n",
    "                (f'{mic_group}_{mic_genus}', 'level_2', mic_name_counts),\n",
    "                (f'{mic_group}_{mic_genus}_{mic_id}', 'level_3', mic_id_counts),\n",
    "            ]\n",
    "\n",
    "            # Adding entries for each level\n",
    "            for node, level, counts in levels:\n",
    "                node_count = counts\n",
    "                new_df = pd.DataFrame([{  # Create a small DataFrame for each row\n",
    "                    'node': node,\n",
    "                    'level': level,\n",
    "                    'counts': counts,\n",
    "                    'node_counts': f'{node} ({node_count})'\n",
    "                }])\n",
    "                transformed_dfs.append(new_df)  # Append to the list of DataFrames\n",
    "    \n",
    "    \n",
    "    else:\n",
    "        return \"focus can only be microbes or metabolites, check spelling and try again\"\n",
    "    \n",
    "\n",
    "    \n",
    "    # Adding the organism row\n",
    "    total_rows = len(df)  # Gets the number of rows in the input dataframe\n",
    "    organ_row = {'node': organ,\n",
    "                 'level': 'level_0',\n",
    "                 'counts': total_rows,\n",
    "                 'node_counts': f'{organ} ({total_rows})'}\n",
    "    \n",
    "    #create the df\n",
    "    organ_df = pd.DataFrame([organ_row])\n",
    "    \n",
    "    #add to df list\n",
    "    transformed_dfs.append(organ_df)\n",
    "    \n",
    "    #combine all dfs\n",
    "    transformed_df = pd.concat(transformed_dfs, ignore_index=True)\n",
    "\n",
    "    # Removing duplicate entries and resetting index\n",
    "    transformed_df = transformed_df.drop_duplicates().reset_index(drop=True)\n",
    "\n",
    "    # add another column label_counts\n",
    "    transformed_df['label_counts'] = transformed_df['node'].str.split('_').str[-1] + ' (' + transformed_df['counts'].astype(str) + ')'\n",
    "\n",
    "    return transformed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "id": "7598440f",
   "metadata": {},
   "outputs": [],
   "source": [
    "brain_metab_nodes = transform_data_sheet(tog6_brain, focus='metabolites', organ='brain')\n",
    "brain_metab_nodes.to_csv('../data/JointRPCA/3xtg_brain_additional_metab-info_Cytoscape.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "id": "50fccc63",
   "metadata": {},
   "outputs": [],
   "source": [
    "brain_micro_nodes = transform_data_sheet(tog6_brain, focus='microbes', organ='brain')\n",
    "brain_micro_nodes.to_csv('../data/JointRPCA/3xtg_brain_additional_micro-info_Cytoscape.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a80ac1b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41489e38",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd9dc44f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "d274163d",
   "metadata": {},
   "outputs": [],
   "source": [
    "tog4_fecal = tog3_fecal[['OGU', 'metab_row_id', 'lowest_taxa_id', 'NPC#class', 'jointRPCA_co-occur_value']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "3f2b1202",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[94], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m t0 \u001b[38;5;241m=\u001b[39m time\u001b[38;5;241m.\u001b[39mtime()\n\u001b[0;32m----> 2\u001b[0m \u001b[43mcreate_graphml_from_df\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtog4_fecal\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mn_clusters\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m10\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpercentile_cutoff\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m75\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m      3\u001b[0m \u001b[43m                         \u001b[49m\u001b[43moutput_file_path\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m../data/JointRPCA/3xtg_fecal_microbe-metabolite_network.graphml\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      4\u001b[0m t1 \u001b[38;5;241m=\u001b[39m time\u001b[38;5;241m.\u001b[39mtime()\n\u001b[1;32m      5\u001b[0m total \u001b[38;5;241m=\u001b[39m t1\u001b[38;5;241m-\u001b[39mt0\n",
      "Cell \u001b[0;32mIn[90], line 39\u001b[0m, in \u001b[0;36mcreate_graphml_from_df\u001b[0;34m(df, output_file_path, n_clusters, percentile_cutoff)\u001b[0m\n\u001b[1;32m     36\u001b[0m row \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m     37\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m tid \u001b[38;5;129;01min\u001b[39;00m metabolite_ids:\n\u001b[1;32m     38\u001b[0m     \u001b[38;5;66;03m# Get correlation value (or other relevant feature)\u001b[39;00m\n\u001b[0;32m---> 39\u001b[0m     correlation \u001b[38;5;241m=\u001b[39m filtered_df[(filtered_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmicrobe_id\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m cid) \u001b[38;5;241m&\u001b[39m (\u001b[43mfiltered_df\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mmetabolite_id\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m==\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mtid\u001b[49m)][\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mcorrelation\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mvalues\n\u001b[1;32m     40\u001b[0m     row\u001b[38;5;241m.\u001b[39mappend(correlation[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(correlation) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;241m0\u001b[39m)  \u001b[38;5;66;03m# Handle missing correlations\u001b[39;00m\n\u001b[1;32m     41\u001b[0m microbe_feature_matrix\u001b[38;5;241m.\u001b[39mappend(row)\n",
      "File \u001b[0;32m~/miniforge3/envs/qiime2-2023.5-gemelli/lib/python3.8/site-packages/pandas/core/ops/common.py:72\u001b[0m, in \u001b[0;36m_unpack_zerodim_and_defer.<locals>.new_method\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     68\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mNotImplemented\u001b[39m\n\u001b[1;32m     70\u001b[0m other \u001b[38;5;241m=\u001b[39m item_from_zerodim(other)\n\u001b[0;32m---> 72\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mmethod\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mother\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/qiime2-2023.5-gemelli/lib/python3.8/site-packages/pandas/core/arraylike.py:42\u001b[0m, in \u001b[0;36mOpsMixin.__eq__\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     40\u001b[0m \u001b[38;5;129m@unpack_zerodim_and_defer\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m__eq__\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     41\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__eq__\u001b[39m(\u001b[38;5;28mself\u001b[39m, other):\n\u001b[0;32m---> 42\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cmp_method\u001b[49m\u001b[43m(\u001b[49m\u001b[43mother\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moperator\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43meq\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/qiime2-2023.5-gemelli/lib/python3.8/site-packages/pandas/core/series.py:6243\u001b[0m, in \u001b[0;36mSeries._cmp_method\u001b[0;34m(self, other, op)\u001b[0m\n\u001b[1;32m   6240\u001b[0m rvalues \u001b[38;5;241m=\u001b[39m extract_array(other, extract_numpy\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, extract_range\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m   6242\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m np\u001b[38;5;241m.\u001b[39merrstate(\u001b[38;5;28mall\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m-> 6243\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m \u001b[43mops\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcomparison_op\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   6245\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_construct_result(res_values, name\u001b[38;5;241m=\u001b[39mres_name)\n",
      "File \u001b[0;32m~/miniforge3/envs/qiime2-2023.5-gemelli/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:287\u001b[0m, in \u001b[0;36mcomparison_op\u001b[0;34m(left, right, op)\u001b[0m\n\u001b[1;32m    284\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m invalid_comparison(lvalues, rvalues, op)\n\u001b[1;32m    286\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m is_object_dtype(lvalues\u001b[38;5;241m.\u001b[39mdtype) \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(rvalues, \u001b[38;5;28mstr\u001b[39m):\n\u001b[0;32m--> 287\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m \u001b[43mcomp_method_OBJECT_ARRAY\u001b[49m\u001b[43m(\u001b[49m\u001b[43mop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrvalues\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    289\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    290\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m _na_arithmetic_op(lvalues, rvalues, op, is_cmp\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "File \u001b[0;32m~/miniforge3/envs/qiime2-2023.5-gemelli/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:75\u001b[0m, in \u001b[0;36mcomp_method_OBJECT_ARRAY\u001b[0;34m(op, x, y)\u001b[0m\n\u001b[1;32m     73\u001b[0m     result \u001b[38;5;241m=\u001b[39m libops\u001b[38;5;241m.\u001b[39mvec_compare(x\u001b[38;5;241m.\u001b[39mravel(), y\u001b[38;5;241m.\u001b[39mravel(), op)\n\u001b[1;32m     74\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 75\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mlibops\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mscalar_compare\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mravel\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43my\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     76\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m result\u001b[38;5;241m.\u001b[39mreshape(x\u001b[38;5;241m.\u001b[39mshape)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "t0 = time.time()\n",
    "create_graphml_from_df(tog4_fecal, n_clusters=10, percentile_cutoff=75,\n",
    "                         output_file_path='../data/JointRPCA/3xtg_fecal_microbe-metabolite_network.graphml')\n",
    "t1 = time.time()\n",
    "total = t1-t0\n",
    "print(\"total hours: \", total/3600)\n",
    "#generally has more than brain\n",
    "#currently using NPC#class, but consider Compound_Name \n",
    "#takes about X hours now"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eac4edff",
   "metadata": {},
   "source": [
    "# Cytoscape\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65761887",
   "metadata": {},
   "source": [
    "1.  File > Import > Network from File.. (select .graphml file)\n",
    "2.  Then, more stuff...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffd6f9f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to find a way to make points farther apart - change \"gravity\"?\n",
    "#maybe ask Helena or Simone?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
